managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/OpAddEntry.java

```diff
@@ -165,7 +165,7 @@ public void addComplete(int rc, final LedgerHandle lh, long entryId, Object ctx)
             return;
         }
 
-        if (ledger.getId() != lh.getId()) {
+        if (lh != null && ledger != null && ledger.getId() != lh.getId()) {
```

Comment:
I'm not sure this would solve the problem as it appears to be a race condition where the object is being recycled.